rosdep/base.yaml

```diff
@@ -3294,6 +3297,7 @@ nodejs-legacy:
 npm:
   debian: [npm]
   fedora: [npm]
+  gentoo: [net-libs/nodejs]
```

Comment:
I can't comment on the usefulness of requiring certain USE flags or not. I think it can be counterproductive for cases where the USE flag controls the building of some feature into it as they normally do (that is, affecting the outcome of a given installed binary or such).
But in case of nodejs, the thing is that when there's no USE=npm, there will be no npm package manager installed at all. Albeit USE=npm is the default, but user could have disabled it and then your rosdep thing doesn't force it back.
So there's USE flags affecting what libraries are linked to (and therefore probably also what features are supported, or what library is used for some feature - e.g ssl), and USE flags that control if some part of the source package is built/installed or not.
Binary distributions simply build the source package and split it into multiple binary packages afterwards, but Gentoo as a source distribution would have to do it at source package level, as that's what we have primarily. Often making the tarball be able to build things separately is too heavy of a maintenance burden to be feasible for the maintainer and then a USE flag is used to control that.